tests/test_tag.py

```diff
@@ -155,3 +156,37 @@ def test_push_tag_failed(mock_repo):
     with patch('git.repo.fun.name_to_object'):
         with pytest.raises(exceptions.PushException):
             repo.tag.push("my_tag", "origin")
+
+
+def test_list_tags(mock_repo):
+    """
+   GIVEN GitRepo is initialized with a path and repo
+   WHEN GitTag.list_tags() is called
+   THEN repo.tags is returned
+   """
+    expected = ['tag1', 'tag2']
+    my_local_mock = mock_repo
+    type(my_local_mock).tags = PropertyMock(return_value=expected)
```

Comment:
This looks off to me, since type() is a function that returns a type object, and we're trying to assign a value to that? Maybe this will be easier to do once list() returns strings. The unit test for mocking remote names may also be of interest, as another way to go about mocking objects.

Reply:
PropertyMocks can't be attached directly to a mock object like mock_repo, instead they need to be attached to they type, please see https://docs.python.org/3/library/unittest.mock.html#unittest.mock.PropertyMock for more information.

I agree it looks odd, however this does allow of the control over what is expected. You could modify the conftest.py to add a 'tags' mock but would not have control over what was returned without also adding additional arguments to the fixture.

Reply:
Right! I think what we did for the remote names with mock.configure() would be an option as well, but there are advantages to this way as well, it's a lot more concise. Thanks for the link!